lib/buffer.js

```diff
@@ -64,7 +64,17 @@ function alignPool() {
  * much breakage at this time. It's not likely that the Buffer constructors
  * would ever actually be removed.
  **/
+var newBufferWarned = false;
 function Buffer(arg, encodingOrOffset, length) {
+  if (!new.target && !newBufferWarned) {
+    newBufferWarned = true;
+    process.emitWarning(
+      'Using Buffer without `new` will soon stop working. ' +
+      'Use `new Buffer`, or preferably ' +
+      'Buffer.from, Buffer.allocUnsafe or Buffer.alloc instead.',
```

Comment:
This might backfire. People might be using `Buffer()` in performance-critical code. They might try `Buffer.alloc` in a hurry, see that it slows down their code, then shrug and just use `new Buffer()`.